streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -714,6 +714,13 @@ void runOnce() {
 
         final long pollLatency = pollPhase();
 
+        // Optimization to skip the rest of the processing loop in case the thread was requested to shut down during
+        // the poll phase
```

Comment:
Good idea but I think we do this a few lines down

Reply:
🤦‍♀️  Oh wow how did I not see that lol. I'll just bump the log to INFO